triton-constraint-circuit/src/lib.rs

```diff
@@ -781,58 +781,56 @@ impl<II: InputIndicator> ConstraintCircuitMonad<II> {
         multicircuit: &[ConstraintCircuitMonad<II>],
         target_degree: isize,
     ) -> usize {
-        // The relevant fields of a ConstraintCircuit.
-        // Avoids interior mutability in a HashSet, which is a foot gun.
-        #[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
-        struct PartialNodeInfo {
-            id: usize,
-            degree: isize,
-        }
-
         assert!(!multicircuit.is_empty());
         let multicircuit = multicircuit
             .iter()
             .map(|c| c.clone().consume())
             .collect_vec();
 
+        // Precalculate all node degrees, to avoid having to recalculate them
+        // multiple times, which would be a slow affair.
+        let node_degrees = Self::all_degrees(&multicircuit);
```

Comment:
I suggest inlining the used-once method `Self::all_degrees()` and shortening the comment.

```suggestion
        // Computing a node's degree is slow; this cache de-duplicates work.
        let node_degrees = Self::all_nodes_in_multicircuit(&multicircuit)
            .into_iter()
            .map(|node| (node.id, node.degree()))
            .collect::<HashMap<_, _>>();
```